src/components/signup/EmailConfirmation.js

```diff
@@ -1,104 +1,196 @@
+    this.setState({ sendingCode: true, code: '', errorMessage: '' })
+    try {
+      await API.sendVerificationEmail(this.props.screenProps.data)
+      this.setState({ sendingCode: false, renderButton: false, resentCode: true }, this.displayDelayedRenderButton)
+      
+      //turn checkmark back into regular resend text
+      setTimeout(() => this.setState({ ...this.state, resentCode: false }), 2000)
+    } catch (e) {
+      log.error(e)
+      this.setState({
+        errorMessage: e.message || e.response.data.message,
+        sendingCode: false,
+        renderButton: true,
+      })
+    }
+  }
+  
+  render() {
+    const { errorMessage, renderButton, loading, code, resentCode } = this.state
+    const { styles } = this.props
+    
+    return (
+      <CustomWrapper handleSubmit={this.handleSubmit} footerComponent={() => <React.Fragment />}>
+        <Section.Stack grow justifyContent="flex-start">
+          <Section.Row justifyContent="center" style={styles.row}>
+            <Section.Title textTransform="none">{'Enter the verification code \n sent to your email'}</Section.Title>
           </Section.Row>
-        </Section>
-        <Wrapper style={styles.containerPadding}>
-          <Section.Row justifyContent="center" grow>
-            <Section.Text fontFamily="slab" fontSize={22} color="surface">
-              {`In order to continue, \n go to your e-mail app and confirm registration`}
-            </Section.Text>
+          <Section.Stack justifyContent="center">
+            <OtpInput
+              shouldAutoFocus
+              numInputs={this.numInputs}
+              onChange={this.handleChange}
+              isInputNum={true}
+              hasErrored={errorMessage !== ''}
+              errorStyle={styles.errorStyle}
+              value={code}
+            />
+            <ErrorText error={errorMessage} />
+          </Section.Stack>
+          <Section.Row alignItems="center" justifyContent="center" style={styles.row}>
+            <CodeAction status={resentCode ? DONE : renderButton ? PENDING : WAIT} handleRetry={this.handleRetry} />
           </Section.Row>
-        </Wrapper>
-      </Section>
-    </CustomWrapper>
+        </Section.Stack>
+        <LoadingIndicator force={loading} />
+      </CustomWrapper>
+    )
+  }
+}
+
+const CodeAction = ({ status, handleRetry }) => {
+  if (status === DONE) {
+    return <Icon size={16} name="success" color="blue" />
+  } else if (status === WAIT) {
+    return (
+      <Section.Text fontFamily="regular" fontSize={14} color="gray80Percent">
+        Please wait a few seconds until the email arrives
+      </Section.Text>
+    )
+  }
+  return (
+    <Section.Text fontFamily="medium" fontSize={14} color="primary" onPress={handleRetry}>
+      Send me the code again
+    </Section.Text>
   )
 }
 
 const getStylesFromProps = ({ theme }) => ({
-  containerPadding: {
-    padding: normalize(28),
-    alignItems: 'center',
+  informativeParagraph: {
+    margin: '1em',
```

Comment:
we use only dp units. no em/rem. also please take extra care to make sure sizes: paddings,fonts,margins match the design

Reply:
This component took from the SmsForm component. You want what I did here through the normalize method?

Reply:
@StanislavShevchenko you should look at the design and use absolute sizes